data/helpers.d/filesystem

```diff
@@ -80,3 +80,49 @@ properly with chmod/chown." >&2
     chmod 755 $TMP_DIR
     echo $TMP_DIR
 }
+
+# Calculate and store a file checksum into the app settings
+#
+# $app should be defined globally when calling this helper
```

Comment:
No it's not needed.
You should consider to a variable can't be limited to the parent script.
A variable, if it set to `local` can't be used out of the function which create it. But, it can be used in any function beneath itself.

So, in your script, if $app is set to `local`, you still can use it in any functions you call. And if you not set it at `local` nor `global`, you can use it the same way.

The matter is only if you set a variable at `local` and try to use it outside this function.